tests/run_tests.py

```diff
@@ -46,14 +50,42 @@ def test_writenek():
 	assert field.var    == fieldw.var
 	assert field.wdsz   == fieldw.wdsz
 	assert (field.time - fieldw.time) < 1e-3
-	assert field.lims.pos.all()  == fieldw.lims.pos.all()
-	assert field.lims.vel.all()  == fieldw.lims.vel.all()
-	assert field.lims.pres.all() == fieldw.lims.pres.all()
-	assert field.lims.scal.all() == fieldw.lims.scal.all()
-	assert np.abs(field.elem[123].vel - fieldw.elem[123].vel).max() < 1e-9
-	assert np.abs(field.elem[-1].pos - fieldw.elem[-1].pos).max() < 1e-9
-	assert np.abs(field.elem[-1].pres - fieldw.elem[-1].pres).max() < 1e-9
+	npt.assert_array_equal(field.lims.pos, fieldw.lims.pos)
+	npt.assert_array_equal(field.lims.vel, fieldw.lims.vel)
+	npt.assert_array_equal(field.lims.pres, fieldw.lims.pres)
+	npt.assert_array_equal(field.lims.scal, fieldw.lims.scal)
+
+	for elem, elemw in zip(field.elem, fieldw.elem):
+		npt.assert_array_equal(elem.pos,  elemw.pos)
+		npt.assert_array_equal(elem.vel,  elemw.vel)
+		npt.assert_array_equal(elem.pres, elemw.pres)
+		npt.assert_array_equal(elem.scal, elemw.scal)
+
+
+def test_writenek_big_endian():
+	import pymech.neksuite as ns
+
+	fname = './tests/nek/channel3D_0.f00001'
+	field = ns.readnek(fname)
+
+	field.endian = "big"
+	fnamew = './test_0_big.f00001'
+	status = ns.writenek(fnamew, field)
+	assert status == 0
+
+	fieldw = ns.readnek(fnamew)
+	assert field.endian == fieldw.endian == "big"
```

Comment:
With the code I just committed this test should fail, but I don't think it's a bug. I think it's okay to let writenek write in whatever endianness it wants, not respecting field.endian, because the check and byte swap can be done in readnek (and is done in Nek5000). As far as I know the reason why this 6.54321 number is written there is because the file can be in little or big endian with no guarantee, and it's read properly whichever way it is.

Reply:
The weird thing is test **do not fail**. I do not understand why.

Reply:
Because the test does not run. You did not call test_writenek_scalars() in main :)

Reply:
I understand your explanation that Nek5000 will take care of the byteswap so writing in one endianness (what is native to the system) is fine for most workflows.

However, I can't help but wonder if someone (or me) would end up with a mixture of files with different endianness and complain in the future. It is best not to deprecate an existing feature.

Reply:
> Because the test does not run.
Ahhh... I will change the filename to `test_all` so that `pytest` will automatically detect all unittests.

Reply:
This is a good point, but I'm not completely convinced. For three reasons:

- I doubt anyone ever used pymech with a big-endian file, let alone *needed* to write specifically a big-endian file,
- as far as I understand, Nek5000 only checks this while reading, and always writes with the native endianness,
- for now, supporting choosing the endianness while writing means much slower writing.

But yes I can revert my changes for now until we find a better solution. I don't know what @jcanton thinks?

Reply:
The test `test_writenek_big_endian` fails now - that is good. I will come back to this later.

Reply:
There is also `ndarray.byteswap()`! I can implement that later.